### PR TITLE
[#9789] Add a loading indicator and button to username input of start…

### LINKED ISSUE #9789: Add a loading indicator and button to username input of start new chat
### User Story

As a user I want feedback when I'm entering a ENS username, when it's looking up my public key and when it's resolved correctly, even on a slower connection. 

### Description

The scope of this issue is 
• Add a dedicated 'Start' button next to the input with three states: disabled (greyed out), pending (greyed out with spinner), ready (blue). this is the same button that's used in ENS name registry.
• We show the disabled state whenever you can't proceed forward, correct state when a correct contact code or ENS username is entered, pending when it's resolving the chat key.
• Change the position of the QR Code button to be placed in the right side of Top Bar. Change the icon to the QR Code to be consistent with wallet.
• Change the subtitle informing of invalid entry to `User not found`
• The container in which this subtitle is shown should always be visible, 16px high, so that the subtitle when shown, doesn't push the contact list downward.
• Match the subtitle style with Figma, it's `font size: 13, font weight: 500, line height: 16, black'

### Acceptance Criteria

All of the above points were implemented and are correct with this [Figma design here ](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=7957%3A169)

### Notes

When in doubt with chat key resolution, please get in touch with @flexsurfer for help
thanks! 🙏

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #9789

### CODE REVIEW

**Review by vitvly — APPROVED** [review]

**Comment by flexsurfer** [comment]
@Serhy or @churik could you take a look?

File: src/status_im/utils/debounce.cljs [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+    (swap! timeout assoc event-key (js/setTimeout #(re-frame/dispatch event) time))))
 
-(defn clear []
-  (when @timeout (js/clearTimeout @timeout)))
-
-(def chill? (atom false))
+(def chill (atom {}))
 
 (defn dispatch-and-chill
   "Dispateches event and ignores next calls in period of *time* ms"
```
Comment (vitvly) [inline src/status_im/utils/debounce.cljs:24]: Not in this PR, but noticed it here: `dispateches` typo.

File: src/status_im/ui/screens/add_new/new_chat/views.cljs [hunk 2]
```diff
@@ -6,53 +6,90 @@
...
+         :preserve-input?     true
+         :accessibility-label :enter-contact-code-input
+         :return-key-type     :go}]]
+      [react/view {:width 16}]
+      [input-icon state]]
+     [react/text {:style styles/message}
+      (cond (= state :error)
+            (i18n/label :t/user-not-found)
+            (= state :valid)
+            (str (when ens-name (str ens-name " • "))
```
Comment (vitvly) [inline src/status_im/ui/screens/add_new/new_chat/views.cljs]: Probably would make sense to move the dot separator to a constant.

**Review by vitvly — APPROVED** [review]

**Comment by Serhy** [comment]
Tested on Android 8.1 and iOS 13.3(iPhoneX) latest builds:

**Issue1**
Can not add contact via QR code scan using new button: 
a) 'Unable to read this code'  when I scan valid public key 

b) app crashes when I scan invalid public key (like a wallet address QR code)
(Android error in that case:     
Screenshot 2020-02-24 at 18 10 41

**Issue2**
When typing chars to find a contact when the informing subtitle appears contacts below jumps on a pixel or two (see below gif)
jumping-contacts


**Issue3**
When I search my own user ENS name or public key there is a "User not found" status of that name. 
**Expected result** (like we have now in release build): `You can't add yourself`

**Comment by flexsurfer** [comment]
thanks @Serhy can confirm that issue1 only in my PR and not in develop? because i didn't change anything related to QR, for issue3 @errorists wdyt ?

**Comment by errorists** [comment]
re: issue 2. this empty space between the input and list where the subtitle appears should have a constant height to avoid any jumps if the text is present / not.

issue 3. absolutely right. tho i remember we wanted to add the ability to start a chat with yourself a long time ago, that's a totally separate discussion. for now `That's you, to start a chat choose someone else` (I'd prefer if it refer as starting a chat in copy rather than adding yourself, since technically you're not adding anyone quite yet)

thanks guys, you're awesome!

**Comment by flexsurfer** [comment]
thanks @Serhy all issues should be fixed now

**Comment by Serhy** [comment]
@flexsurfer, everything works great now but I can still reproduce Issue2 on Android 8.1 / Android9. It's fixed on iOS side, but on Android list of user jumps whenever informing subtitle appears (latest build tested https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200225-120611-bbbe0b-pr-universal.apk)

**Comment by flexsurfer** [comment]
@Serhy fixed

**Comment by flexsurfer** [comment]
@Serhy fixed

**Comment by Serhy** [comment]
Works greatly for me with iOS and Android now!

**Pull request merged**